src/App.jsx

```diff
@@ -14,45 +16,129 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedArray(goods, { sortField, reversed }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          if (good1 > good2) {
+            return 1;
+          }
+
+          if (good1 < good2) {
+            return -1;
+          }
+
+          return 0;
+
+        case SORT_FIELD_LENGTH:
+          if (good1.length > good2.length) {
+            return 1;
+          }
+
+          if (good1.length < good2.length) {
+            return -1;
+          }
+
+          return 0;
+
+        default: return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
```

Comment:
Rename boolean variable according [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig). Fix all cases if present
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```